core/src/test/java/kafka/admin/ConfigCommandUnitTest.java

```diff
@@ -410,6 +448,430 @@ public void testOptionEntityTypeNames() {
+                assertEquals(expectedAlterEntity, alteration.entity());
+                Collection<ClientQuotaAlteration.Op> ops = alteration.ops();
+                assertEquals(expectedAlterOps, new HashSet<>(ops));
+                alteredConfigs.set(true);
+                return alterResult;
+            }
+        };
+        ConfigCommand.alterConfig(mockAdminClient, createOpts);
+        assertTrue(describedConfigs.get());
+        assertTrue(alteredConfigs.get());
+    }
+
+    @Test
+    public void testAlterIpConfig() {
+        Map.Entry<List<String>, Map<String, String>> t = toValues(Optional.of("1.2.3.4"), ClientQuotaEntity.IP);
+        List<String> singleIpArgs = t.getKey();
+        Map<String, String> singleIpEntry = t.getValue();
+        ClientQuotaEntity singleIpEntity = new ClientQuotaEntity(singleIpEntry);
+        t = toValues(Optional.of(""), ClientQuotaEntity.IP);
+        List<String> defaultIpArgs = t.getKey();
+        Map<String, String> defaultIpEntry = t.getValue();
+        ClientQuotaEntity defaultIpEntity = new ClientQuotaEntity(defaultIpEntry);
+
+        List<String> deleteArgs = Arrays.asList("--delete-config", "connection_creation_rate");
+        Set<ClientQuotaAlteration.Op> deleteAlterationOps = new HashSet<>(Collections.singletonList(new ClientQuotaAlteration.Op("connection_creation_rate", null)));
+        Map<String, Double> propsToDelete = Collections.singletonMap("connection_creation_rate", 50.0);
+
+        List<String> addArgs = Arrays.asList("--add-config", "connection_creation_rate=100");
+        Set<ClientQuotaAlteration.Op> addAlterationOps = new HashSet<>(Collections.singletonList(new ClientQuotaAlteration.Op("connection_creation_rate", 100.0)));
+
+        verifyAlterQuotas(concat(singleIpArgs, deleteArgs), singleIpEntity, propsToDelete, deleteAlterationOps);
+        verifyAlterQuotas(concat(singleIpArgs, addArgs), singleIpEntity, Collections.emptyMap(), addAlterationOps);
+        verifyAlterQuotas(concat(defaultIpArgs, deleteArgs), defaultIpEntity, propsToDelete, deleteAlterationOps);
+        verifyAlterQuotas(concat(defaultIpArgs, addArgs), defaultIpEntity, Collections.emptyMap(), addAlterationOps);
+    }
+
+    @Test
+    public void shouldAddClientConfig() {
+        List<String> alterArgs = Arrays.asList("--add-config", "consumer_byte_rate=20000,producer_byte_rate=10000",
+            "--delete-config", "request_percentage");
+        Map<String, Double> propsToDelete = Collections.singletonMap("request_percentage", 50.0);
+
+        Set<ClientQuotaAlteration.Op> alterationOps = new HashSet<>(Arrays.asList(
+            new ClientQuotaAlteration.Op("consumer_byte_rate", 20000d),
+            new ClientQuotaAlteration.Op("producer_byte_rate", 10000d),
+            new ClientQuotaAlteration.Op("request_percentage", null)
+        ));
+
+        KafkaFuture.BiConsumer<Optional<String>, Optional<String>> verifyAlterUserClientQuotas = (userOpt, clientOpt) -> {
```

Comment:
Maybe we can use `java.util.function.BiConsumer` ?

Reply:
Yes. Fxied.